src/js/components/DataChart/Detail.js

```diff
@@ -88,55 +88,70 @@ const Detail = ({
+            ({ property }) =>
+              ((!activeProperty || activeProperty === property) &&
+                data?.[index]?.[property] !== undefined) ||
+              (axis && axis.x && axis.x.property === property),
+          )
+          .map((serie) => {
+            const axisValue = horizontalProp
+              ? data[index][serie.property]
+              : index;
+            return `${serie.label || serie.property} ${renderValue(
+              serie,
+              axisValue,
+            )}`;
+          })
+          .join(' ');
+      }
+      return undefined;
+    },
+    [activeProperty, axis, data, horizontalProp, renderValue, series],
+  );
+
   useEffect(() => {
     if (detailIndex !== undefined) {
-      const content = series
-        .filter(
-          ({ property }) =>
-            ((!activeProperty || activeProperty === property) &&
-              data?.[detailIndex]?.[property] !== undefined) ||
-            (axis && axis.x && axis.x.property === property),
-        )
-        .map((serie) => {
-          const axisValue = horizontalProp
-            ? data[detailIndex][serie.property]
-            : detailIndex;
-          return `${serie.label || serie.property} ${renderValue(
-            serie,
-            axisValue,
-          )}`;
-        })
-        .join(' ');
-      announce(content);
+      announce(getContent(detailIndex), 'assertive');
     }
-  }, [
-    activeProperty,
-    announce,
-    axis,
-    data,
-    detailIndex,
-    horizontalProp,
-    renderValue,
-    series,
-  ]);
+  }, [announce, detailIndex, getContent]);
 
   return (
     <>
       <Keyboard
-        onLeft={() => {
-          if (detailIndex === undefined) setDetailIndex(data.length - 1);
-          else if (detailIndex > 0) setDetailIndex(detailIndex - 1);
+        onLeft={(event) => {
+          event.preventDefault();
+          if (detailIndex === undefined) {
+            setDetailIndex(data.length - 1);
+            announce(getContent(data.length - 1), 'assertive');
+          } else if (detailIndex > 0) {
+            setDetailIndex(detailIndex - 1);
+            announce(getContent(detailIndex - 1), 'assertive');
+          }
         }}
-        onRight={() => {
-          if (detailIndex === undefined) setDetailIndex(0);
-          else if (detailIndex < data.length - 1)
+        onRight={(event) => {
+          event.preventDefault();
+          if (detailIndex === undefined) {
+            setDetailIndex(0);
+            announce(getContent(0), 'assertive');
+          } else if (detailIndex < data.length - 1) {
             setDetailIndex(detailIndex + 1);
+            announce(getContent(detailIndex + 1), 'assertive');
+          }
         }}
       >
         <DetailControl
           key="band"
-          tabIndex={0}
           fill
+          role="region"
+          tabIndex={0}
+          aria-label={`
+            Data chart: Use the left and right arrow keys to navigate the data.
+          `}
```

Comment:
Whenever you make a generic element, like `div`, focusable, you must also override the generic role and add an accessible name/label to the element.

Reply:
I like this change, makes it much clearer when the focus lands on this div.

To allow this aria-label to be internationalized can we have this string come from the `languages/default.json` file and then use `MessageContext`

Reply:
Sure. No problem.

Reply:
Done! I've added `dataChart.detailTitle` and `dataChart.detailFocus` with a default title for the chart and keyboard instructions to announce when `Detail` has focus.